operators/src/processing/point_in_polygon.rs

```diff
@@ -0,0 +1,775 @@
+use futures::stream::BoxStream;
+use futures::{StreamExt, TryStreamExt};
+use snafu::ensure;
+
+use geoengine_datatypes::collections::{
+    MultiPointCollection, MultiPolygonCollection, VectorDataType,
+};
+use geoengine_datatypes::primitives::{Coordinate2D, TimeInterval};
+
+use crate::adapters::FeatureCollectionChunkMerger;
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+    Operator, QueryContext, QueryProcessor, QueryRectangle, TypedVectorQueryProcessor,
+    VectorOperator, VectorQueryProcessor, VectorResultDescriptor,
+};
+use crate::error;
+use crate::util::Result;
+use arrow::array::BooleanArray;
+use itertools::Itertools;
+
+pub type PointInPolygonFilter = Operator<()>;
+
+#[typetag::serde]
+impl VectorOperator for PointInPolygonFilter {
+    fn initialize(
+        self: Box<Self>,
+        context: &ExecutionContext,
+    ) -> Result<Box<InitializedVectorOperator>> {
+        ensure!(
+            self.vector_sources.len() == 2,
+            error::InvalidNumberOfVectorInputs {
+                expected: 2..3,
+                found: self.vector_sources.len()
+            }
+        );
+        ensure!(
+            self.raster_sources.is_empty(),
+            error::InvalidNumberOfRasterInputs {
+                expected: 0..1,
+                found: self.raster_sources.len()
+            }
+        );
+
+        let vector_sources = self
+            .vector_sources
+            .into_iter()
+            .map(|o| o.initialize(context))
+            .collect::<Result<Vec<Box<InitializedVectorOperator>>>>()?;
+
+        ensure!(
+            vector_sources[0].result_descriptor().data_type == VectorDataType::MultiPoint,
+            error::InvalidType {
+                expected: VectorDataType::MultiPoint.to_string(),
+                found: vector_sources[0].result_descriptor().data_type.to_string(),
+            }
+        );
+        ensure!(
+            vector_sources[1].result_descriptor().data_type == VectorDataType::MultiPolygon,
+            error::InvalidType {
+                expected: VectorDataType::MultiPolygon.to_string(),
+                found: vector_sources[1].result_descriptor().data_type.to_string(),
+            }
+        );
+
+        Ok(InitializedPointInPolygonFilter::new(
+            (),
+            vector_sources[0].result_descriptor(),
+            vec![],
+            vector_sources,
+            (),
+        )
+        .boxed())
+    }
+}
+
+pub type InitializedPointInPolygonFilter = InitializedOperatorImpl<(), VectorResultDescriptor, ()>;
+
+impl InitializedOperator<VectorResultDescriptor, TypedVectorQueryProcessor>
+    for InitializedPointInPolygonFilter
+{
+    fn query_processor(&self) -> Result<TypedVectorQueryProcessor> {
+        let point_processor = if let TypedVectorQueryProcessor::MultiPoint(point_processor) =
```

Comment:
`self.vector_sources[0].query_processor()?.multi_point().unwrap()` would be shorter

Reply:
358aa07 

Yeah, these shortcuts came after the initial implementation, but are super handy